File: server/src/com/broll/gainea/server/core/goals/impl/e1/G_AllIslands.java
package com.broll.gainea.server.core.goals.impl.e1;

import com.broll.gainea.server.core.goals.OccupyGoal;
import com.broll.gainea.server.core.goals.GoalDifficulty;
import com.broll.gainea.server.core.map.impl.GaineaMap;

public class G_AllIslands extends OccupyGoal {
    public G_AllIslands() {
        super(GoalDifficulty.MEDIUM, "Erobere alle Inseln");
    }

    @Override
    protected void initOccupations() {
        occupy(GaineaMap.Islands.values());
    }

}

File: server/src/main/com/broll/gainea/server/core/goals/impl/e1/G_AllIslands.kt
package com.broll.gainea.server.core.goals.impl.e1

import com.broll.gainea.server.core.goals.GoalDifficulty
import com.broll.gainea.server.core.goals.OccupyGoal
import com.broll.gainea.server.core.map.impl.GaineaMap

class G_AllIslands : OccupyGoal(GoalDifficulty.MEDIUM, "Erobere alle Inseln") {
    override fun initOccupations() {
        occupy(*GaineaMap.Islands.entries.toTypedArray())
    }
}
